crates/fuel-core/src/graphql_api/indexation/coins_to_spend.rs

```diff
@@ -0,0 +1,739 @@
+        Event::ForcedTransactionFailed { .. } => Ok(()),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use fuel_core_storage::{
+        iter::IterDirection,
+        transactional::WriteTransaction,
+        StorageAsMut,
+    };
+    use fuel_core_types::{
+        fuel_tx::{
+            Address,
+            AssetId,
+        },
+        services::executor::Event,
+    };
+    use rand::seq::SliceRandom;
+
+    use itertools::Itertools;
+    use proptest::{
+        collection::vec,
+        prelude::*,
+    };
+
+    use crate::{
+        database::{
+            database_description::off_chain::OffChain,
+            Database,
+        },
+        graphql_api::{
+            indexation::{
+                coins_to_spend::{
+                    update,
+                    RETRYABLE_BYTE,
+                },
+                error::IndexationError,
+                test_utils::{
+                    make_coin,
+                    make_nonretryable_message,
+                    make_retryable_message,
+                },
+            },
+            storage::coins::{
+                CoinsToSpendIndex,
+                CoinsToSpendIndexKey,
+            },
+        },
+    };
+
+    use super::NON_RETRYABLE_BYTE;
+
+    fn assert_index_entries(
+        db: &Database<OffChain>,
+        expected_entries: &[(Address, AssetId, [u8; 1], u64)],
+    ) {
+        let actual_entries: Vec<_> = db
+            .entries::<CoinsToSpendIndex>(None, IterDirection::Forward)
+            .map(|entry| entry.expect("should read entries"))
+            .map(|entry| {
+                (
+                    entry.key.owner(),
+                    entry.key.asset_id(),
+                    [entry.key.retryable_flag()],
+                    entry.key.amount(),
+                )
+            })
+            .collect();
+
+        assert_eq!(expected_entries, actual_entries.as_slice());
+    }
+
+    #[test]
+    fn coins_to_spend_indexation_enabled_flag_is_respected() {
+        use tempfile::TempDir;
+        let tmp_dir = TempDir::new().unwrap();
+        let mut db: Database<OffChain> =
+            Database::open_rocksdb(tmp_dir.path(), None, Default::default(), 512)
+                .unwrap();
+        let mut tx = db.write_transaction();
+
+        // Given
+        const COINS_TO_SPEND_INDEX_IS_DISABLED: bool = false;
+        let base_asset_id = AssetId::from([0; 32]);
+
+        let owner_1 = Address::from([1; 32]);
+        let owner_2 = Address::from([2; 32]);
+
+        let asset_id_1 = AssetId::from([11; 32]);
+        let asset_id_2 = AssetId::from([12; 32]);
+
+        let coin_1 = make_coin(&owner_1, &asset_id_1, 100);
+        let coin_2 = make_coin(&owner_1, &asset_id_2, 200);
+        let message_1 = make_retryable_message(&owner_1, 300);
+        let message_2 = make_nonretryable_message(&owner_2, 400);
+
+        // TODO[RC]: No clone() required for coins? Double check the types used,
```

Comment:
What does RC mean?

Reply:
Right, nice! I typically reference the issue ID I'm working on: `//TODO(<issue-number>)`. It also makes it fast to find todos relating to follow-ups :)